src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -106,188 +107,195 @@ void EnCow_Init(Actor* thisx, PlayState* play) {
     } else {
         Actor_OfferGetItem(&this->actor, play, GI_MILK, 10000.0f, 100.0f);
     }
 }
 
-void func_809DF7D8(EnCow* this, PlayState* play) {
+void EnCow_GiveMilk(EnCow* this, PlayState* play) {
     if ((Message_GetState(&play->msgCtx) == TEXT_STATE_EVENT) && Message_ShouldAdvance(play)) {
         this->actor.flags &= ~ACTOR_FLAG_16;
         Message_CloseTextbox(play);
-        this->actionFunc = func_809DF778;
+        this->actionFunc = EnCow_GiveMilkWait;
         Actor_OfferGetItem(&this->actor, play, GI_MILK, 10000.0f, 100.0f);
     }
 }
 
-void func_809DF870(EnCow* this, PlayState* play) {
+void EnCow_CheckForEmptyBottle(EnCow* this, PlayState* play) {
     if ((Message_GetState(&play->msgCtx) == TEXT_STATE_EVENT) && Message_ShouldAdvance(play)) {
         if (Inventory_HasEmptyBottle()) {
             Message_ContinueTextbox(play, 0x2007);
-            this->actionFunc = func_809DF7D8;
+            this->actionFunc = EnCow_GiveMilk;
         } else {
             Message_ContinueTextbox(play, 0x2013);
-            this->actionFunc = func_809DF6BC;
+            this->actionFunc = EnCow_TalkEnd;
         }
     }
 }
 
-void func_809DF8FC(EnCow* this, PlayState* play) {
+void EnCow_Talk(EnCow* this, PlayState* play) {
     if (Actor_ProcessTalkRequest(&this->actor, play)) {
-        this->actionFunc = func_809DF870;
+        this->actionFunc = EnCow_CheckForEmptyBottle;
     } else {
         this->actor.flags |= ACTOR_FLAG_16;
         func_8002F2CC(&this->actor, play, 170.0f);
         this->actor.textId = 0x2006;
     }
-    func_809DF494(this, play);
+    EnCow_UpdateAnimation(this, play);
 }
 
-void func_809DF96C(EnCow* this, PlayState* play) {
+void EnCow_Idle(EnCow* this, PlayState* play) {
     if ((play->msgCtx.ocarinaMode == OCARINA_MODE_00) || (play->msgCtx.ocarinaMode == OCARINA_MODE_04)) {
-        if (DREG(53) != 0) {
-            if (this->unk_276 & 4) {
-                this->unk_276 &= ~0x4;
-                DREG(53) = 0;
+        if (R_PLAYED_EPONAS_SONG) {
+            if (this->cowFlags & EN_COW_FLAG_WONT_GIVE_MILK) {
+                this->cowFlags &= ~EN_COW_FLAG_WONT_GIVE_MILK;
+                R_PLAYED_EPONAS_SONG = false;
             } else {
                 if ((this->actor.xzDistToPlayer < 150.0f) &&
-                    (ABS((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) < 0x61A8)) {
-                    DREG(53) = 0;
-                    this->actionFunc = func_809DF8FC;
+                    (ABS((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) < 25000)) {
+                    R_PLAYED_EPONAS_SONG = false;
+                    this->actionFunc = EnCow_Talk;
                     this->actor.flags |= ACTOR_FLAG_16;
                     func_8002F2CC(&this->actor, play, 170.0f);
                     this->actor.textId = 0x2006;
                 } else {
-                    this->unk_276 |= 4;
+                    this->cowFlags |= EN_COW_FLAG_WONT_GIVE_MILK;
```

Comment:
I'm confused by the `EN_COW_FLAG_WONT_GIVE_MILK` flag, what effect/purpose does it have in game?

(`msgCtx.ocarinaMode` docs courtesy of @engineer124 : https://github.com/zeldaret/oot/compare/master...engineer124:oot:ocarina )
`OCARINA_MODE_00` is `OCARINA_MODE_NONE` 

so most of the time `this->cowFlags &= ~EN_COW_FLAG_WONT_GIVE_MILK;` will run and the flag will be cleared :thinking:

Reply:
>  If this flag was enabled on the next tick along with R_PLAYED_EPONAS_SONG, it would mean that there wasnt another actor who could handle the epona's song reg flag, so it turns both of them off.

Aaaah! :brain: thanks that makes a lot of sense
I'm not sure how to name the flag then, could you write a brief comment somewhere (near `cowFlags & EN_COW_FLAG_WONT_GIVE_MILK` maybe) about what it achieves?